### PR TITLE
Dsl2

### PR DESCRIPTION
**Pull request by MaxUlysse**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Still working on it, but managed to fix so far:
- MarkDuplicates
- BamQC
- target_bed
- building indices from path
- using already built indices from path

File: main.nf [hunk 1]
```diff
@@ -197,32 +197,33 @@ params.snpeff_db               = params.genome ? params.genomes[params.genome].s
...
-target_bed        =  params.target_bed        ?: Channel.empty()
-vep_cache         = params.vep_cache         ?: Channel.empty()
+cadd_indels       = params.cadd_indels       ? file(params.cadd_indels)      : file("${params.outdir}/no_file")
+cadd_indels_tbi   = params.cadd_indels_tbi   ? file(params.cadd_indels_tbi)  : file("${params.outdir}/no_file")
+cadd_wg_snvs      = params.cadd_wg_snvs      ? file(params.cadd_wg_snvs)     : file("${params.outdir}/no_file")
+cadd_wg_snvs_tbi  = params.cadd_wg_snvs_tbi  ? file(params.cadd_wg_snvs_tbi) : file("${params.outdir}/no_file")
+pon               = params.pon               ? file(params.pon)              : file("${params.outdir}/no_file")
+snpeff_cache      = params.snpeff_cache      ? file(params.snpeff_cache)     : file("${params.outdir}/no_file")
+target_bed        = params.target_bed        ? file(params.target_bed)       : file("${params.outdir}/no_file")
+vep_cache         = params.vep_cache         ? file(params.vep_cache)        : file("${params.outdir}/no_file")
```
Comment (FriederikeHanssen) [inline main.nf:227]: Does the no_file folder then remain empty?

File: main.nf [hunk 2]
```diff
@@ -354,14 +355,14 @@ workflow {
...
+    bwa = params.bwa ? file(params.bwa) : BUILD_INDICES.out.bwa
+    dbsnp_tbi = params.dbsnp ? params.dbsnp_index ? file(params.dbsnp_index) : BUILD_INDICES.out.dbsnp_tbi : Channel.empty()
+    dict = params.dict ? file(params.dict) : BUILD_INDICES.out.dict
+    fai = params.fasta_fai ? file(params.fasta_fai) : BUILD_INDICES.out.fai
+    germline_resource_tbi = params.germline_resource ? params.germline_resource_index ? file(params.germline_resource_index) : BUILD_INDICES.out.germline_resource_tbi : Channel.empty()
     intervals = BUILD_INDICES.out.intervals
-    known_indels_tbi = params.known_indels ? params.known_indels_index ?: BUILD_INDICES.out.known_indels_tbi.collect() : Channel.empty()
-    pon_tbi = params.pon ? params.pon_index ?: BUILD_INDICES.out.pon_tbi : Channel.empty()
+    known_indels_tbi = params.known_indels ? params.known_indels_index ? file(params.known_indels_index) : BUILD_INDICES.out.known_indels_tbi.collect() : Channel.empty()
+    pon_tbi = params.pon ? params.pon_index ? file(params.pon_index) : BUILD_INDICES.out.pon_tbi : Channel.empty()
```
Comment (FriederikeHanssen) [inline main.nf:366]: We should indent this at some point so it looks as nice as the rest :)

File: main.nf [hunk 3]
```diff
@@ -422,22 +423,21 @@ workflow {
     bam_mapped = bam_single.mix(MERGE_BAM_MAPPED(bam_multiple))
     //if(save_bam_mapped || !(params.known_indels))
     //TODO: https://github.com/nf-core/sarek/blob/bce378e09de25bb26c388b917f93f84806d3ba27/main.nf#L1478
-    //But  if SAMTOOLS_INDEX is not run, markduplicates does not work
-         bam_mapped = SAMTOOLS_INDEX_MAPPED(bam_mapped)
+    //But if SAMTOOLS_INDEX is not run, markduplicates does not work
+    bam_mapped = SAMTOOLS_INDEX_MAPPED(bam_mapped)
```
Comment (FriederikeHanssen) [inline main.nf:427]: This comment is obsolet with your fix, right?

File: main.nf [hunk 4]
```diff
@@ -464,31 +464,55 @@ workflow {
         }
 
         GATHERBQSRREPORTS(recaltable)
-        // if ('baserecalibrator' in skip_qc) baseRecalibratorReport.close()
+        table = GATHERBQSRREPORTS.out.table
```
Comment (FriederikeHanssen) [inline main.nf:467]: rename to bqsr_table?

File: main.nf [hunk 5]
```diff
@@ -464,31 +464,55 @@ workflow {
...
-    bamqc = BWAMEM2_MEM.out//.mix(MERGE_BAM_RECAL.out)
-    //if(!('bamqc' in skipQC))
-        BAMQC(bamqc)//, target_bed)
+    if (!('samtools' in skip_qc)) {
+        SAMTOOLS_STATS(BWAMEM2_MEM.out.mix(recal))
+    }
 
+    if (!('bamqc' in skip_qc)) {
+        BAMQC(BWAMEM2_MEM.out.mix(recal), target_bed)
+    }
```
Comment (FriederikeHanssen) [inline main.nf:515]: Maybe for consistency we could always set the output to a variable, so above somewhere bwamem2_out = BWAMEM2_MEM.out ..... Maybe a more descriptive variable name.

File: main.nf (hunk 1 above)
Comment (MaxUlysse) [reply main.nf:227]: I'm actually creating the `no_file` file a couple of lines before with:
```
file("${params.outdir}/no_file").text = "no_file\n"
```

File: main.nf (hunk 2 above)
Comment (MaxUlysse) [reply main.nf:366]: Good point

File: main.nf (hunk 3 above)
Comment (MaxUlysse) [reply main.nf:427]: Wasn't sure about it, but probably

File: main.nf (hunk 4 above)
Comment (MaxUlysse) [reply main.nf:467]: Yes, that's a good idea

File: main.nf (hunk 5 above)
Comment (MaxUlysse) [reply main.nf:515]: That's a good idea too.
Maybe something like `bwamem2_bam` or `mapped_bam`

File: main.nf (hunk 1 above)
Comment (FriederikeHanssen) [reply main.nf:227]: Ah ok, so actually just an empty file is used then?

**Pull request merged**